semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/http/HttpSemanticConventionUtils.java

```diff
@@ -196,4 +257,161 @@ public static boolean isValidUrl(String url) {
+
+    if (event.getAttributes() == null || event.getAttributes().getAttributeMap() == null) {
+      return Optional.empty();
+    }
+
+    Map<String, AttributeValue> attributeValueMap = event.getAttributes().getAttributeMap();
+    for (String useragent : USER_AGENT_ATTRIBUTES) {
+      if (attributeValueMap.get(useragent) != null
+          && !StringUtils.isEmpty(attributeValueMap.get(useragent).getValue())) {
+        return Optional.of(attributeValueMap.get(useragent).getValue());
+      }
+    }
+    return Optional.empty();
+  }
+
+  public static Optional<String> getHttpUserAgentFromHeader(Event event) {
+
+    if (event.getAttributes() == null || event.getAttributes().getAttributeMap() == null) {
+      return Optional.empty();
+    }
+
+    Map<String, AttributeValue> attributeValueMap = event.getAttributes().getAttributeMap();
+    String userAgentKey = RawSpanConstants.getValue(HTTP_USER_AGENT_REQUEST_HEADER);
+    if (attributeValueMap.get(userAgentKey) != null
+        && !StringUtils.isEmpty(attributeValueMap.get(userAgentKey).getValue())) {
+      return Optional.of(attributeValueMap.get(userAgentKey).getValue());
+    }
+
+    return Optional.empty();
+  }
+
+  public static Optional<String> getHttpHost(Event event) {
+
+    if (event.getAttributes() == null || event.getAttributes().getAttributeMap() == null) {
+      return Optional.empty();
+    }
+
+    Map<String, AttributeValue> attributeValueMap = event.getAttributes().getAttributeMap();
+    for (String host : HOST_ATTRIBUTES) {
+      if (attributeValueMap.get(host) != null) {
+        return Optional.of(attributeValueMap.get(host).getValue());
+      }
+    }
+    return Optional.empty();
+  }
+
+  public static Optional<String> getHttpPath(Event event) {
+
+    if (event.getAttributes() == null || event.getAttributes().getAttributeMap() == null) {
+      return Optional.empty();
+    }
+
+    Map<String, AttributeValue> attributeValueMap = event.getAttributes().getAttributeMap();
+    for (String path : URL_PATH_ATTRIBUTES) {
+      if (attributeValueMap.get(path) != null) {
+        String s = attributeValueMap.get(path).getValue();
+        if (StringUtils.isNotBlank(s) && s.startsWith(SLASH)) {
+          return Optional.of(s);
+        }
+      }
+    }
+    return Optional.empty();
+  }
+
+  public static Optional<String> getHttpMethod(Event event) {
+
+    if (event.getAttributes() == null || event.getAttributes().getAttributeMap() == null) {
+      return Optional.empty();
+    }
+
+    Map<String, AttributeValue> attributeValueMap = event.getAttributes().getAttributeMap();
+    for (String method : METHOD_ATTRIBUTES) {
+      if (attributeValueMap.get(method) != null
+          && !StringUtils.isEmpty(attributeValueMap.get(method).getValue())) {
+        return Optional.of(attributeValueMap.get(method).getValue());
+      }
+    }
+    return Optional.empty();
+  }
+
+  public static Optional<String> getHttpScheme(Event event) {
```

Comment:
Scheme can be set from this method as well https://github.com/hypertrace/hypertrace-ingester/blob/b97d519115b02b5992c069b62685ef0c838e5532/span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/fieldgenerators/HttpFieldsGenerator.java#L536